src/NServiceBus.RabbitMQ/Receiving/MessagePump.cs

```diff
@@ -79,7 +79,26 @@ public void Start(PushRuntimeSettings limitations)
             connection = connectionFactory.CreateConnection($"{settings.InputQueue} MessagePump");
 
             var channel = connection.CreateModel();
-            channel.BasicQos(0, (ushort)Math.Min(ushort.MaxValue, Convert.ToUInt16(limitations.MaxConcurrency) * prefetchCountPerMessageProcessor), false);
+
+            int prefetchMultiplier = 3;
+            long prefetchCount;
+
+            if (overriddenPrefetchCount > 0)
+            {
+                prefetchCount = overriddenPrefetchCount;
+
+                if (prefetchCount < maxConcurrency)
+                {
+                    Logger.Warn($"The specified prefetch count '{prefetchCount}' is smaller than the specified maximum concurrency '{maxConcurrency}'. The maximum concurrency value will be used as the prefetch count instead.");
+                    prefetchCount = maxConcurrency;
+                }
+            }
+            else
+            {
+                prefetchCount = (long)maxConcurrency * prefetchMultiplier;
+            }
+
+            channel.BasicQos(0, (ushort)prefetchCount, false);
```

Comment:
Wont the cast to `ushort` throw an exception if the value is > `ushort.MaxValue`?

This is why we originally had `(ushort)Math.Min(ushort.MaxValue,...`.